### ISSUE TITLE
Text is disappearing from the Autocomplete input when typing fast.

### ISSUE THREAD

**Issue by vivanenko** [vikramlearning/blazorbootstrap#407]
Text is disappearing from the Autocomplete input when typing fast.

Steps to reproduce the behaviour:
1. Go to https://demos.blazorbootstrap.com/autocomplete
2. Type fast any text in any autocomplete input
4. See this strange behaviour. The same bug is reproduced in my Blazor project

 - BlazorBootstrap: 1.10.3
 - Blazor Server
 - .NET Version: .NET 7

**Comment by gvreddy04** [comment]
@vivanenko Thank you for trying **BlazorBootstrap** components. What are your expectations from your testing team? How many characters does the user type in your case? Do you think the user will type that many characters in the search bar in real time?

**Comment by vivanenko** [comment]
> @vivanenko Thank you for trying **BlazorBootstrap** components. What are your expectations from your testing team? How many characters does the user type in your case? Do you think the user will type that many characters in the search bar in real time?

In our typical use case a user enters one word (a name) in a second. Let's take "Daina" as an example. When I type it in less than a second the text is partially disappearing and appearing again with some delay. Sometimes the Autocomplete swallows characters so they are completely lost. Both issues confuse users.

**Comment by MarvinKlein1508** [comment]
For me this is also happening when typing in normal speed. I think the best way to fix this is to only search after a certain delay after the last keystroke and not on every keystroke.

**Comment by MarvinKlein1508** [comment]
@gvreddy04 what is the point of the delay in the first place? It works fine for me without any delay.

**Comment by gvreddy04** [comment]
@vivanenko  @MarvinKlein1508 We should not remove the delay. There is a specific reason for the delay.

Here is the scenario: If a user is typing fast (i.e., 3-4 characters per second), you are making 3-4 calls to the API or database to filter the data. If you are simply filtering on the existing collection, which is already available, it will still not filter until there is a 300-millisecond delay in the next keystroke from the user.

I'll investigate on the text disappearing.

**Comment by gvreddy04** [comment]
@vivanenko @MarvinKlein1508 Another option I can give you is the Debounce parameter. If it is not required in your case, you can set it to 0.

**Comment by MarvinKlein1508** [comment]
> @vivanenko @MarvinKlein1508 Another option I can give you is the Debounce parameter. If it is not required in your case, you can set it to 0.

I would see a bool value here so get rid of the await in the first place when you don't need it. Speeds up the application by removing the state machine

**Comment by gvreddy04** [comment]
> @vivanenko @MarvinKlein1508 We should not remove the delay. There is a specific reason for the delay.
> 
> Here is the scenario: If a user is typing fast (i.e., 3-4 characters per second), you are making 3-4 calls to the API or database to filter the data. If you are simply filtering on the existing collection, which is already available, it will still not filter until there is a 300-millisecond delay in the next keystroke from the user. Same approach is used by all the AutoComplete components.
> 
> I'll investigate on the text disappearing.

**Video 1:** Without debouncing 300 milliseconds, it will make more number of calls to the API as the user types in.


https://github.com/vikramlearning/blazorbootstrap/assets/2337067/81a70459-78ac-4f35-8608-a456d6bc8248

**Comment by gvreddy04** [comment]
> > @vivanenko @MarvinKlein1508 We should not remove the delay. There is a specific reason for the delay.
> > Here is the scenario: If a user is typing fast (i.e., 3-4 characters per second), you are making 3-4 calls to the API or database to filter the data. If you are simply filtering on the existing collection, which is already available, it will still not filter until there is a 300-millisecond delay in the next keystroke from the user. Same approach is used by all the AutoComplete components.
> > I'll investigate on the text disappearing.
> 
> ## Video 1: Without debouncing 300 milliseconds, it will make more number of calls to the API as the user types in.
> 
>  blazorbootstrap_issue_407.mp4

## Video 2: Wuth default 300 milli seconds debouncing


https://github.com/vikramlearning/blazorbootstrap/assets/2337067/cbe641fd-803c-4074-8482-053bda3b04bf

**Comment by gvreddy04** [comment]
@vivanenko @MarvinKlein1508 Were you able to reproduce the issue at https://demos.blazorbootstrap.com/autocomplete?

If so, can you share a recording of how to reproduce it? I am currently unable to reproduce the issue myself.

**Comment by MarvinKlein1508** [comment]
yes you can reproduce this easily on demos. Will send you video soon.

**Comment by MarvinKlein1508** [comment]
Here you go. I was typing Bandita very fast
![bandita](https://github.com/vikramlearning/blazorbootstrap/assets/32510006/c05c56ef-bb13-4b39-8611-a614202f9c77)

**Comment by gvreddy04** [comment]
@MarvinKlein1508 Thank you for your time.

**Comment by MarvinKlein1508** [comment]
@gvreddy04 I just took another look on it. This issue only happens for me in the demos. When I run both demo apps locally then it works fine for me.

**Comment by pdavisbdbc** [comment]
Is this still being reviewed? The issue does not happen locally, but it does happen in the Demo and when an application is deployed where latency between the client and the server is higher.

**Comment by danbopes** [comment]
I get the same issue
